thrall/app/controllers/ReaperController.scala

```diff
@@ -46,35 +44,25 @@ class ReaperController(
     override val persistenceIdentifier: String = config.persistenceIdentifier
   }
 
-  config.maybeReaperBucket.foreach { reaperBucket =>
-    scheduler.scheduleAtFixedRate(
-      initialDelay = 0 seconds,
-      interval = INTERVAL,
-    ){ () =>
-      if(store.client.doesObjectExist(reaperBucket, CONTROL_FILE_NAME)) {
-        logger.info("Reaper is paused")
-        es.countTotalSoftReapable(isReapable).map(metrics.softReapable.increment(Nil, _).run)
-        es.countTotalHardReapable(isReapable).map(metrics.hardReapable.increment(Nil, _).run)
-      } else {
-        es.countImagesIngestedInLast(ONE_WEEK)(DateTime.now(DateTimeZone.UTC)).flatMap { imagesIngestedInLast7Days =>
-
-          val imagesIngestedPer15Mins = imagesIngestedInLast7Days / REAPS_PER_WEEK
-
-          val countOfImagesToReap = Math.min(imagesIngestedPer15Mins, 1000).toInt
-
-          if (countOfImagesToReap == 1000) {
-            logger.warn(s"Reaper is reaping at maximum rate of 1000 images per $INTERVAL. If this persists, the INTERVAL will need to become more frequent.")
-          }
-
+  (config.maybeReaperBucket, config.maybeReaperCountPerRun) match {
+    case (Some(reaperBucket), Some(countOfImagesToReap)) =>
+      scheduler.scheduleAtFixedRate(
+        initialDelay = 0 seconds,
+        interval = INTERVAL,
+      ){ () =>
+        if(store.client.doesObjectExist(reaperBucket, CONTROL_FILE_NAME)) {
+          logger.info("Reaper is paused")
+          es.countTotalSoftReapable(isReapable).map(metrics.softReapable.increment(Nil, _).run)
+          es.countTotalHardReapable(isReapable).map(metrics.hardReapable.increment(Nil, _).run)
+        } else {
           val deletedBy = "reaper"
-
           Future.sequence(Seq(
             doBatchSoftReap(countOfImagesToReap, deletedBy),
             doBatchHardReap(countOfImagesToReap, deletedBy)
           ))
         }
       }
-    }
+    case _ => ()
```

Comment:
worth logging that reaper won't run due to missing config?

Reply:
good suggestion - done